core/src/main/java/com/microsoft/applicationinsights/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -0,0 +1,194 @@
+    private final Object lock = new Object();
+
+    public NetworkStatsbeat(TelemetryClient telemetryClient, long interval) {
+        super(telemetryClient, interval);
+    }
+
+    @Override
+    protected void send() {
+        String instrumentation = String.valueOf(getInstrumentation());
+
+        if (requestSuccessCount.get() != 0) {
+            MetricTelemetry requestSuccessCountSt = createStatsbeatTelemetry(REQUEST_SUCCESS_COUNT, requestSuccessCount.get());
+            requestSuccessCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(requestSuccessCountSt);
+            logger.debug("send a NetworkStatsbeat {}: {}", REQUEST_SUCCESS_COUNT, requestSuccessCountSt);
+        }
+
+        if (requestFailureCount.get() != 0) {
+            MetricTelemetry requestFailureCountSt = createStatsbeatTelemetry(REQUEST_FAILURE_COUNT, requestFailureCount.get());
+            requestFailureCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(requestFailureCountSt);
+            logger.debug("send a NetworkStatsbeat {}: {}", REQUEST_FAILURE_COUNT, requestFailureCountSt);
+        }
+
+        double durationAvg = getRequestDurationAvg();
+        if (durationAvg != 0) {
+            MetricTelemetry requestDurationSt = createStatsbeatTelemetry(REQUEST_DURATION, durationAvg);
+            requestDurationSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(requestDurationSt);
+            logger.debug("send a NetworkStatsbeat {}: {}", REQUEST_DURATION, requestDurationSt);
+        }
+
+        if (retryCount.get() != 0) {
+            MetricTelemetry retryCountSt = createStatsbeatTelemetry(RETRY_COUNT, retryCount.get());
+            retryCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(retryCountSt);
+            logger.debug("send a NetworkStatsbeat {}: {}", RETRY_COUNT, retryCountSt);
+        }
+
+        if (throttlingCount.get() != 0) {
+            MetricTelemetry throttleCountSt = createStatsbeatTelemetry(THROTTLE_COUNT, throttlingCount.get());
+            throttleCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(throttleCountSt);
+            logger.debug("send a NetworkStatsbeat {}: {}", THROTTLE_COUNT, throttleCountSt);
+        }
+
+        if (exceptionCount.get() != 0) {
+            MetricTelemetry exceptionCountSt = createStatsbeatTelemetry(EXCEPTION_COUNT, exceptionCount.get());
+            exceptionCountSt.getProperties().put(CUSTOM_DIMENSIONS_INSTRUMENTATION, instrumentation);
+            telemetryClient.track(exceptionCountSt);
+            logger.debug("send a NetworkStatsbeat{}: {}", EXCEPTION_COUNT, exceptionCountSt);
+        }
+    }
+
+    @Override
+    protected synchronized void reset() {
```

Comment:
i think there's a decent size race condition between send and reset, another option is to have a static `current` or `active` instance, and throw that away each time, that narrows the race condition down to these two lines (which we can decide to synchronize or not)

```
NetworkStatsbeat local = active
active = new NetworkStatsbeat()
```
then send `local`